ui/component/or-app/src/index.ts

```diff
@@ -419,9 +437,71 @@ export class OrApp<S extends AppStateKeyed> extends LitElement {
             }
         } else if(event === OREvent.ONLINE) {
             if(this._offline) {
-                this._store.dispatch((setOffline(false)))
+                this._showOfflineFallback = false;
+                this._completeOfflineFallbackTimer(); // complete fallback timer
+                this._store.dispatch((setOffline(false)));
+            }
+        } else if(event === OREvent.RECONNECT_FAILED) {
+            this._startOfflineFallbackTimer(); // start fallback timer (if not done yet)
+
+        } else if(event === OREvent.CONSOLE_VISIBLE) {
+            this._store.dispatch((setVisibility(true)));
+
+            // When the manager appears on Mobile devices, but the connection is OFFLINE,
+            // we reset the timer to the {appConfig.offlineTimeout} seconds. This is because we saw issues with reopening the app,
+            // and seeing a connection interval of 30+ seconds. We now give the user the benefit of the doubt, by resetting the timer.
+            if(manager.console?.isMobile && this._offline) {
+                this._startOfflineFallbackTimer(true);
+                manager.reconnect();
             }
+
+        } else if(event === OREvent.CONSOLE_HIDDEN) {
+            this._store.dispatch((setVisibility(false)));
+        }
+    }
+
+    // Offline timer logic
+    //
+    // This will start a Deferred promise that keeps track of the 'wait before showing offline page' timer.
+    // - Resolving the promise updates the 'show offline fallback' variable based on OFFLINE state.
+    // - Rejecting the promise 'aborts the process' and skips that logic and does nothing.
+    //
+    // To explain; when the Manager reports "We're offline!" it will wait 10+ seconds before visually reporting the user that he/she is offline.
```

Comment:
Why do we need to 10s delay in showing offline state in the UI?

Reply:
For UI / UX purposes we didn't want to lock the UI when a user is only offline for 0.5 seconds.
So I implemented a minimum of 10 seconds, before it is shown to the user.
In practice this works fine for us.